staging/src/k8s.io/kube-apiextensions-server/pkg/controller/finalizer/crd_finalizer.go

```diff
@@ -114,29 +114,49 @@ func (c *CRDFinalizer) sync(key string) error {
 		return err
 	}
 
-	// Its possible for a naming conflict to have removed this resource from the API after instances were created.
-	// For now we will cowardly stop finalizing.  If we don't go through the REST API, weird things may happen:
-	// no audit trail, no admission checks or side effects, finalization would probably still work but defaulting
-	// would be missed.  It would be a mess.
-	// This requires human intervention to solve, update status so they have a reason.
-	// TODO split coreNamesAccepted from extendedNamesAccepted.  If coreNames were accepted, then we have something to cleanup
-	// and the endpoint is serviceable.  if they aren't, then there's nothing to cleanup.
-	if !apiextensions.IsCRDConditionFalse(crd, apiextensions.NameConflict) {
+	// Now we can start deleting items.  We should use the REST API to ensure that all normal admission runs.
+	// Since we control the endpoints, we know that delete collection works. No need to delete if not established.
+	if apiextensions.IsCRDConditionTrue(crd, apiextensions.Established) {
+		if err := c.deleteInstances(crd); err != nil {
+			apiextensions.SetCRDCondition(crd, apiextensions.CustomResourceDefinitionCondition{
```

Comment:
Indeed. Didn't notice that you have a special reason value for a failed deletion.